### ISSUE TITLE
Generator for Predictions

### ISSUE THREAD

**Issue by dalekube** [metal3d/keras-video-generators#7]
I am able to apply the VideoFrameGenerator class to train my neural network with sequences of frames, but I'm struggling to apply it to unseen data to make predictions. The videos in the unseen data set do not have classes; the goal is to predict the classes. I'm having to create a new generator from scratch for this purpose, one that matches the VideoFrameGenerator.

I think the VideoFrameGenerator class would be very powerful if it can also be used for unseen videos that do not have labels/classes.

**Comment by metal3d** [comment]
Hi, thanks a lot to use this package.

I don't understand the problem. If you need to have a generator to get unseen videos, you can either:

- change the `glob_pattern` to hit every videos in subdirectories (e.g. `./videos/**/{classname}/*`
- put the video in a single directory and use the generator that will give you a single `classname`

Then use the generator to get them one by one (generators respect the iterator interface, so you can use `for video in my_generator` - is that what you're looking for ?

**Comment by dalekube** [comment]
You suggest I can put a video in a subdirectory by it's class. What if you don't know the class? What if you deploy your model and need to use the generator to process new videos that have not been classified (aka you need to make a prediction from the unseen data)?

I don't think your suggestion applies to this scenario. You'll have to create a new generator altogether to match how you processed videos during the training phase.

**Comment by metal3d** [comment]
Hi, sorry for the delay,

The generator is made for training. If you want to process videos for inference, you only need to pick frames (+resize) with OpenCV for example.

If you really need to pick frames on a batch of videos with the generator, you can point it to the "unseen" subdirectory and use `for v in the_generator` as suggested in my previous comment.

For example, if I've got files in "my/data/unseen/" so I do:


```
gen = VideoFrameGenerator(
    classes=['unseen'], # "unseen" is where my unseen videos reside
    glob_pattern='my/data/{classname}/*', # I use the generator to find only unseen videos
    nb_frames=NBFRAME,
    split=None, 
    shuffle=False,
    batch_size=BS,
    target_shape=SIZE,
    nb_channel=3, # or 1 for grayscale
    transformation=None, # default is None
    use_frame_cache=True) # or False
```

Then:

```
for v, _ in gen:
   # v contains the batch of video frames
   for b in v:
     # b is "one batch of N frames"
```

Acutally, that works absolutly as any other generator for Keras.

**Comment by metal3d** [comment]
If your videos are in several subdirectories:

```python
gen = VideoFrameGenerator(
    classes=['unseen'], # "unseen" is where my unseen videos reside
    glob_pattern='my/data/{classname}/**/*', # I use the generator to find only unseen videos
    nb_frames=NBFRAME,
    split=None, 
    shuffle=False,
    batch_size=BS,
    target_shape=SIZE,
    nb_channel=3, # or 1 for grayscale
    transformation=None, # default is None
    use_frame_cache=False) # no need cache
```

Note the glob pattern that can visit each subdirectory (`**`).

**Comment by dalekube** [comment]
Thank you for the explanation. I see how that would work now!
